include/wx/generic/maskededit.h

```diff
@@ -0,0 +1,681 @@
+
+    //The real control
+    wxControl* msk_control;
+
+    //Bell on error. Default is true
+    bool m_bellOnError;
+
+    //The mask identifiers
+    enum
+    {
+        mDigitReq = 1, //'#'
+        mDigitOpt,     //'9'
+        mAlphaReq,     //'A'
+        mAlphaOpt,     //'a'
+        mAlpNumReq,    //'N'
+        mAlpNumOpt,    //'n'
+        mAnyCharReq,   //'C'
+        mAnyCharOpt,   //'c'
+        mHexReq,       //'H'
+        mHexOpt,       //'H'
+        mLiteral,
+        mUppChar = 100, //'>'
+        mLowChar = 200  //'<'
+    };
+};
+
+// ----------------------------------------------------------------------------
+// Masked edit controls
+// ----------------------------------------------------------------------------
+
+#if wxUSE_TEXTCTRL
+//wxMaskedEditText is only for wxTextCtrl
+class WXDLLIMPEXP_ADV wxMaskedEditText : public wxMaskedEdit,
+                                         public wxTextCtrl
+{
+public:
+    //Constructors
+    wxMaskedEditText()
+    {
+        mskInit( 0 );
+    }
+
+    wxMaskedEditText(wxWindow *parent, wxWindowID id,
+                     const wxString& value = wxEmptyString,
+                     const wxPoint& pos = wxDefaultPosition,
+                     const wxSize& size = wxDefaultSize,
+                     long style = 0,
+                     const wxValidator& validator = wxDefaultValidator,
+                     const wxString& name = wxTextCtrlNameStr)
+         : wxTextCtrl(parent, id, value, pos, size, style, validator, name)
+    {
+        mskInit( style );
+        mskPostInit();
+    }
+
+    bool Create(wxWindow *parent, wxWindowID id,
+                const wxString& value = wxEmptyString,
+                const wxPoint& pos = wxDefaultPosition,
+                const wxSize& size = wxDefaultSize,
+                long style = 0,
+                const wxValidator& validator = wxDefaultValidator,
+                const wxString& name = wxTextCtrlNameStr)
+    {
+        mskInit( style );
+
+        if ( ! wxTextCtrl::Create(parent, id, value, pos, size, style,
+                                  validator, name) )
+        {
+            return false;
+        }
+        mskPostInit();
+        return true;
+    }
+
+    virtual ~wxMaskedEditText() {};
+
+    //Override SetValue()
+    virtual void SetValue(const wxString& value)
+    {
+        wxTextCtrl::SetValue( SetMaskedValue(value) );
+        //Test the new value and set colours
+        UpdateMaskedColours();
+    }
+
+    //Override ChangeValue()
+    virtual void ChangeValue(const wxString& value)
+    {
+        wxTextCtrl::ChangeValue( SetMaskedValue(value) );
+        //Test the new value and set colours
+        UpdateMaskedColours();
+    }
+
+private:
+    void mskInit( long style )
+    {
+        //Only single line control
+        if ( style & wxTE_MULTILINE )
+        {
+            wxFAIL_MSG("Multiline wxMaskedEditText not allowed yet");
+        }
+    }
+
+    void mskPostInit()
```

Comment:
```suggestion
    void MaskPostInit()
```
This function shouldn't be inline.

Reply:
@vadz ,
Name is changed
But I don't know about inlining it.....